src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs

```diff
@@ -113,16 +113,27 @@ public bool TailCall
         /// <returns>A delegate containing the compiled version of the lambda.</returns>
         public Delegate Compile()
         {
-#if FEATURE_CORECLR
+#if FEATURE_COMPILE
             return Compiler.LambdaCompiler.Compile(this);
 #else
             return new System.Linq.Expressions.Interpreter.LightCompiler().CompileTop(this).CreateDelegate();
 #endif 
         }
 
-#if FEATURE_CORECLR
+#if FEATURE_INTERPRET
+        /// <summary>
+        /// Produces a delegate that represents the lambda expression.
+        /// </summary>
+        /// <returns>A delegate containing the compiled version of the lambda.</returns>
+        public Delegate Interpret()
+        {
```

Comment:
We discussed this a bit before (I've lost where I placed the comments, perhaps I forgot to publish..).
It seems better if the entry point has shape   "Compile(bool preferInterpretation)"

The reasons are:
1) Compile()   does not necessary compile. It basically prefers compiling if available, but otherwise will interpret.
2) The method does not really start interpretation. Technically it still compiles into a delegate, just with a different form that will be executed by the means of interpretation.